src/block_cache/block_chunkcache.c

```diff
@@ -926,6 +1020,61 @@ __wt_chunkcache_reconfig(WT_SESSION_IMPL *session, const char **cfg)
     return (0);
 }
 
+/*
+ * __wt_chunkcache_create_from_metadata --
+ *     Create a new chunk from some stored properties, and link it to the relevant chunk data (on
+ *     disk).
+ */
+int
+__wt_chunkcache_create_from_metadata(WT_SESSION_IMPL *session, const char *name, uint32_t id,
+  wt_off_t file_offset, uint64_t cache_offset, size_t chunk_size)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *newchunk;
+    WT_CHUNKCACHE_HASHID hash_id;
+    WT_DECL_RET;
+    size_t bit_index;
+    uint64_t bucket_id;
+
+    chunkcache = &S2C(session)->chunkcache;
+
+    if (!F_ISSET(chunkcache, WT_CHUNKCACHE_CONFIGURED))
+        return (0);
+
+    /* Sanity check the offset. */
+    WT_ASSERT(session, sizeof(uint64_t) == sizeof(wt_off_t));
+    WT_ASSERT(session, (wt_off_t)cache_offset >= 0);
+
+    bucket_id = __chunkcache_tmp_hash(chunkcache, &hash_id, name, id, file_offset);
+
+    __wt_spin_lock(session, WT_BUCKET_LOCK(chunkcache, bucket_id));
+    WT_ERR(__create_and_populate_chunk(
+      session, &newchunk, file_offset, chunk_size, &hash_id, bucket_id));
+
+    /* Get the position of a specific bit index and link the chunk and its memory cached on disk */
+    bit_index = cache_offset / chunkcache->chunk_size;
+    WT_ASSERT_ALWAYS(session, !__set_bit_index(session, bit_index),
+      "the link between chunk memory and cached data cannot be established as the link is already "
+      "in place");
+    newchunk->chunk_memory = chunkcache->memory + cache_offset;
+
+    TAILQ_INSERT_HEAD(WT_BUCKET_CHUNKS(chunkcache, bucket_id), newchunk, next_chunk);
+    WT_PUBLISH(newchunk->valid, true);
+
+    __wt_verbose_debug2(session, WT_VERB_CHUNKCACHE,
+      "created and linked chunk: %s(%u), offset=%" PRId64 ", size=%lu", (char *)name, id,
```

Comment:
This could use some more comments in general.

Reply:
Updated [97ba0ee](https://github.com/wiredtiger/wiredtiger/pull/9759/commits/97ba0ee1e02fa21eb47178c4d55dca94c405711b)